File: 5-longest-palindromic-substring/solution.py
# Runtime: 274ms, Beats 85.14% of users with Python3
# Memory: 16.66MB, Beats 73.50% of users with Python3

class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        maxLength=0
        substring=""
        for i in range(n):
            l,r=i,i
            while l>=0 and r<n and s[l] == s[r]:
                if r-l+1>maxLength:
                    maxLength=r-l+1
                    substring=s[l:r+1]
                l-=1
                r+=1
            l,r=i,i+1
            while l>=0 and r<n and s[l] == s[r]:
                if r-l+1 > maxLength:
                    maxLength=r-l+1
                    substring=s[l:r+1]
                l-=1
                r+=1
        return substring

File: 5-longest-palindromic-substring/solution.cpp
// Runtime: 8ms, Beats 92.46% of users with C++
// Memory: 8.08MB, Beats 76.44% of users with C++


class Solution {
public:
    string longestPalindrome(string s) {
        int n=s.size();
        int maxi=0;
        int start=0;
        int l,r;
        string substring="";
        for (int i=0;i<n;i++)
        {
            l=i,r=i;
            while (l>=0 && r<n && s[l] == s[r])
            {
                if (r-l+1 > maxi)
                {
                    start=l;
                    maxi=r-l+1;
                }
                l--;
                r++;
            }
            l=i,r=i+1;
            while (l>=0 && r<n && s[l] == s[r])
            {
                if(r-l+1 > maxi)
                {
                    start=l;
                    maxi=r-l+1;
                }
                l--;
                r++;
            }
        }
        return s.substr(start,maxi);
    }
};